zap/src/irgen/mod.rs

```diff
@@ -1,11 +1,29 @@
 #![allow(dead_code)]
+use std::collections::HashMap;
 use std::{fmt::Display, vec};
 
 use crate::config::{NumTy, Range, Ty};
 
 pub mod des;
 pub mod ser;
 
+pub trait VarOccurrencesProvider {
+	fn get_var_occurrences(&mut self) -> &mut HashMap<String, i32>;
+	fn add_occurrence(&mut self, name: String) -> i32 {
+		match self.get_var_occurrences().get(&name) {
+			Some(occurrences) => {
+				let occurrences_inc = occurrences + 1;
+				self.get_var_occurrences().insert(name, occurrences_inc);
+				occurrences_inc
+			}
+			None => {
+				self.get_var_occurrences().insert(name, 1);
+				1
+			}
```

Comment:
If you follow the above question where it returns the value, if this `None` clause is reached, it should just return the var name.
IE: `len_1` -> `len`.

Reply:
That can create shadowing again, like with the header.
`self.push_line("local len = buffer.len(buff)");`

Reply:
oh okay! no worries then